plugins/woocommerce/src/Admin/API/Reports/Products/DataStore.php

```diff
@@ -432,17 +432,51 @@ public static function sync_order_products( $order_id ) {
 		$decimals       = wc_get_price_decimals();
 		$round_tax      = 'no' === get_option( 'woocommerce_tax_round_at_subtotal' );
 
+		$is_full_refund = false;
+		// Order fully refunded won't have the order line items.
+		if ( 'shop_order_refund' === $order->get_type() && empty( $order_items ) ) {
+			$is_full_refund  = true;
+			$parent_order_id = $order->get_parent_id();
+			$parent_order    = wc_get_order( $parent_order_id );
+			$order_items     = $parent_order->get_items();
+
+			// Remove the partial refunded order item if exists.
+			$delete_query = "
+				DELETE product_lookup
+				FROM {$table_name} AS product_lookup
+				INNER JOIN {$wpdb->prefix}wc_order_stats AS order_stats
+					ON order_stats.order_id = product_lookup.order_id
+				WHERE 1 = 1
+					AND order_stats.parent_id = %d
+					AND order_stats.total_sales < 0
+			";
```

Comment:
Thanks. Instead of removing them first, I changed it to select the already existing partial refunds in lookup table, then exclude them from the parent orders line items. 1d54d24